lib/wibox/layout/fixed.lua

```diff
@@ -110,6 +110,14 @@ function fixed:get_children()
     return self.widgets
 end
 
+--- Replace the layout children
+-- @tparam table children A table composed of valid widgets
+function fixed:set_children(children)
+    if not children then return self:reset() end
```

Comment:
I agree with the inconsistency... Perhaps make `get_children()` return them as a "list", too? (Callers could get a table via `{ foo:get_children() }`)
This would again simplify the implementation in most widgets and fixed can use `unpack()`, but this is quite a minor simplification, so I'm not sure if this makes a lot of sense. Feel free to say "No!".

About "I am just being nice for users": Well, is it really nice? Assuming the user meant to do something and e.g. just had a typo in the variable name, the code will silently do nothing instead of causing a Lua error whose backtrace hopefully points at the typo. I'm not sure if "being nice" is always the right thing.